src/onedrive.rs

```diff
@@ -269,56 +270,101 @@ impl OneDrive {
             .await
     }
 
-    /// Create a new folder under an `DriveItem`
-    ///
-    /// Create a new folder [`DriveItem`][drive_item] with a specified parent item or path.
+    /// Create a new [`DriveItem`][drive_item] allowing to set supported attributes.
+    /// [`DriveItem`][drive_item] resources have facets modeled as properties that provide data
+    /// about the [`DriveItem`][drive_item]'s identities and capabilities. You must provide one
+    /// of the following facets to create an item: `bundle`, `file`, `folder`, `remote_item`.
     ///
     /// # Errors
-    /// Will result in `Err` with HTTP `409 CONFLICT` if [`conflict_behavior`][conflict_behavior]
+    /// * Will result in `Err` with HTTP `409 CONFLICT` if [`conflict_behavior`][conflict_behavior]
     /// is set to [`Fail`][conflict_fail] and the target already exists.
+    /// * Will result in `Err` with HTTP `400 BAD REQUEST` facets are not properly set.
     ///
     /// # See also
-    /// [Microsoft Docs](https://docs.microsoft.com/en-us/graph/api/driveitem-post-children?view=graph-rest-1.0)
     ///
+    /// [Microsoft Docs](https://learn.microsoft.com/en-us/onedrive/developer/rest-api/api/driveitem_post_children?view=graph-rest-1.0)
+    ///
+    /// [with_opt]: #method.create_folder_with_option
     /// [drive_item]: ./resource/struct.DriveItem.html
     /// [conflict_behavior]: ./option/struct.DriveItemPutOption.html#method.conflict_behavior
     /// [conflict_fail]: ./enum.ConflictBehavior.html#variant.Fail
-    pub async fn create_folder_with_option<'a>(
+    pub async fn create_drive_item<'a>(
         &self,
         parent_item: impl Into<ItemLocation<'a>>,
-        name: &FileName,
+        drive_item: DriveItem,
         option: DriveItemPutOption,
     ) -> Result<DriveItem> {
+        if drive_item.bundle.is_none()
+            && drive_item.file.is_none()
+            && drive_item.folder.is_none()
+            && drive_item.remote_item.is_none()
+        {
+            // Locally preventing a Bad Request. Other errors are still possible
+            return Err(Error::from_error_response(
+                reqwest::StatusCode::BAD_REQUEST,
+                ErrorResponse {
+                    code: "invalidRequest".to_string(),
+                    message: "Must provide one of the following facets to create an item: Bundle, File, Folder, RemoteItem".to_string(),
+                    inner_error: None,
+                },
+            ));
+        }
```

Comment:
My intent was to add `assert!` to panic immediately because it's clearly a coding error, rather than something that *might* happen during a normal workflow as an `Err` return.

Well, in the other hand, hard-checking here assumes our check is the same as the upstream restrictions. But there may be more restrictions or new facets added in the future.

Thus I think it may be better to just remove these check and let upstream decide. This *should* be enough since the documentation already mentions it. Sorry for the inconvenience.

Reply:
I agree, and don't worry. I've removed the code already, so please let me know of any other changes you might see to improve the code.